### DISCUSSION TITLE
RTK Query dependent queries

### DISCUSSION THREAD

**Discussion by christopanayotovvia** [reduxjs/redux-toolkit#1683] [Q&A]
Hi, 

[The documentation ](https://redux-toolkit.js.org/rtk-query/comparison#comparing-feature-sets)states that the RTK Query supports **dependent query** . I cannot seem to find how to use dependent queries anywhere else. 

I'm little bit how to achieve the following:
Let's say I want to achieve the following:
1. Load `configuration` . In it there is `baseUrl`
2. With the `baseUrl` I want to fetch the user. So the `user` is dependent query on the `configuration`
3. With all other additional request, use data from both `configuration` and `user` ( ex. prepend some headers like x-conf-smth, x-user-smth )

How this can be achieved?

**Comment by phryneas** [comment]
Dependent queries is a term that we have taken from the comparison list on React-Query's homepage, and over there it means that two queries within a component can depend on each other.

```js
const result1 = useFirstQuery(someArg)
const result2 = useSecondQuery(result1.data?.something, { skip: !result1.isFulfilled })
```

**Reply by jalooc** [reply]
For comparison with RQ it does make sense indeed. But the RQ's approach to dependent queries is very shallow, cause it happens outside of RQ itself: in a component's render function. So it's not really an RQ's characteristic - they've done close to nothing to enable _dependent queries_.

What I was kind of hoping to see when hearing _dependent queries_ in context of RTQ, was a way to specify the dependencies _within the RTQ world_. That would probably entail making dependant queries reactive in respect to their dependent state. This would allow to encapsulate entire query's logic inside a query function, just like RTQ already encapsulates nicely all the other parts. It would help mitigate leaking query's logic to the react-land, limiting the need of tackling react limitations like those of hooks etc. - just like Redux has always been doing.

What do you think about such development direction?

**Reply by markerikson** [reply]
It's something we've occasionally thought about as an idea, but have no specific plans or API design atm.

At the moment, the closest you'd get would be either triggering another query inside of an `onQueryStarted` lifecycle after the first query is fulfilled and dispatching a thunk for the second query, or writing a `queryFn` that dispatches both thunks.
